tests/testthat/test-extendr_function_options.R

```diff
@@ -70,3 +70,10 @@ test_that(
     )
   }
 )
+
+
+test_that(
+  "`rust_function()` does not emit any messages when `quiet = TRUE`", {
+    expect_no_message(rust_function(code = "fn func() {}", quiet = TRUE))
+  }
+)
```

Comment:
@Ilia-Kosenkov new test to verify that no messages are output when `quiet = TRUE`